Automattic/newspack-plugin
# feat: campaign group filter with preview

### All Submissions:






### Changes proposed in this Pull Request:

Adds preview functionality to the Campaigns Wizard Campaigns tab. At some point this functionality may make the Preview tab unnecessary, however the new approach doesn't support selecting multiple Campaign Groups so I'm leaving it untouched for now.

Dependent PRs:
~https://github.com/Automattic/newspack-plugin/pull/764~
~https://github.com/Automattic/newspack-plugin/pull/767~
~https://github.com/Automattic/newspack-plugin/pull/768~
~https://github.com/Automattic/newspack-plugin/pull/769~
~https://github.com/Automattic/newspack-plugin/pull/770~

<img width="1446" alt="Screen Shot 2021-01-01 at 9 37 19 AM" src="https://user-images.githubusercontent.com/1477002/103440608-fea1b700-4c14-11eb-84a4-31eb307e1764.png">

### How to test the changes in this Pull Request:

1. Be sure you have created some Campaign Groups and Segments.
2. Navigate to Campaign Wizard, first tab. 
3. Select a Campaign Group. Observe the "Segment to preview" `SelectControl` is shown. 
4. Select a Segment and click Preview, observe normal preview functionality.
5. Compare various preview scenarios in Preview and Campaigns tabs, verify the results are identical.

### Other information:
